### PR TITLE
Resolve warnings in GitHub action workflows + set up ITs

### PR DESCRIPTION
**Pull request by lealobanov**
## Description

- Resolve warnings and deprecated code in GitHub Actions workflows
- Setup test module for integration tests with a dummy test
- Update GH workflows to run integration tests and publish test results
- Unit tests + unit tests results publishing task only runs if build succeeded (avoids redundant runs of this job)
- IT tests + IT tests results publishing task only runs if unit tests succeeded (avoids redundant runs of this job)
- Refactored existing ITs into new test module so they run separately from unit tests

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-jvm-sdk/blob/master/CONTRIBUTING.md#styleguides).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: .github/workflows/ci-publish-release.yml [hunk 1]
```diff
@@ -24,13 +24,14 @@ jobs:
         uses: battila7/get-version-action@v2
 
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
```
Comment (lealobanov) [inline .github/workflows/ci-publish-release.yml:27]: Updated to more recent versions of these actions as they were throwing deprecated warnings

File: .github/workflows/ci-publish-release.yml [hunk 2]
```diff
@@ -39,19 +40,35 @@ jobs:
         run: chmod +x ./gradlew
 
       - name: Build
-        run: ./gradlew --warning-mode all check build -x test
+        id: build
+        run: ./gradlew --warning-mode all check build -x test -x integrationTest
```
Comment (lealobanov) [inline .github/workflows/ci-publish-release.yml:44]: Do not run unit tests and integration tests tasks during initial build

File: .github/workflows/ci-publish-release.yml [hunk 3]
```diff
@@ -39,19 +40,35 @@ jobs:
...
 
-      - name: Test
+      - name: Run Unit Tests
+        id: unit_test
+        if: ${{ steps.build.outcome == 'success' }}
```
Comment (lealobanov) [inline .github/workflows/ci-publish-release.yml:48]: Only trigger unit tests if the initial build succeeded

File: .github/workflows/ci-publish-release.yml [hunk 4]
```diff
@@ -39,19 +40,35 @@ jobs:
...
 
       - name: Publish Unit Test Results
         uses: EnricoMi/publish-unit-test-result-action@v1.18
         if: always()
         with:
-          files: "**/test-results/**/*.xml"
+          check_name: "Unit Test Results"
+          files: "**/test-results/test/**/*.xml"
+
+      - name: Run Integration Tests
```
Comment (lealobanov) [inline .github/workflows/ci-publish-release.yml:60]: Add a separate task for running integration tests, only triggers if unit tests succeeded

File: .github/workflows/ci-publish-release.yml [hunk 5]
```diff
@@ -39,19 +40,35 @@ jobs:
...
+        if: ${{ steps.unit_test.outcome == 'success' }}
+        run: |
+          export PATH="$HOME/.local/bin:$PATH"
+          ./gradlew --no-daemon --max-workers=2 --warning-mode=all integrationTest --continue --stacktrace
+
+      - name: Publish Integration Test Results
```
Comment (lealobanov) [inline .github/workflows/ci-publish-release.yml:66]: Output integration tests results separately

File: build.gradle.kts [hunk 6]
```diff
@@ -78,6 +78,43 @@ tasks.named<KotlinCompile>("compileTestKotlin") {
...
+val intTestRuntimeOnly by configurations.getting
+
+configurations["intTestRuntimeOnly"].extendsFrom(configurations.runtimeOnly.get())
+
+dependencies {
+    intTestImplementation("org.junit.jupiter:junit-jupiter:5.10.1")
+    intTestRuntimeOnly("org.junit.platform:junit-platform-launcher")
+}
+
+val integrationTest = task<Test>("integrationTest") {
```
Comment (lealobanov) [inline build.gradle.kts:101]: Create gradle task for ITs

File: .github/workflows/ci-publish-release.yml (hunk 2 above)
Comment (lealobanov) [reply .github/workflows/ci-publish-release.yml:44]: I also removed an if-clause that allowed one to run the action (PR, snapshot, and release) without unit tests given a matching commit message

**Review by franklywatson — APPROVED** [review]

**Pull request merged**